### PR TITLE
initial work to migrate to typescript

### PR DESCRIPTION
**Pull request by EcZachly**
This is the initial work needed to move the project over to using TypeScript. I have moved most of the project over with this. There's still some JavaScript that needs to be moved and some more standards we need to follow but this is a great first start. 

You'll need to `npm install` `ts-node` and `typescript` globally if you haven't yet.

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+//Middlewares
 import analyticsMiddleware from './lib/middleware/site_traffic_middleware';
 
+//Routers
 import api_router from './routes/api_router';
 import view_router from './routes/view_router';
-import bodyParser from 'body-parser';
-
-let app = express();
+let app: Application = express();
```
Comment (jakewilson801) [inline app.ts:17]: `const`

File: lib/api_wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as CallOfDutyAPI from 'call-of-duty-api';
+import Bluebird from 'bluebird';
+let EMAIL = process.env.WARZONE_EMAIL;
+let PASSWORD = process.env.WARZONE_PASSWORD;
+
+export default class ApiWrapper {
```
Comment (jakewilson801) [inline lib/api_wrapper.ts:6]: Does this have to be a class with a singleton?

File: lib/etl/scrape.js [hunk 3]
```diff
@@ -108,7 +108,7 @@ async function executePipeline(gamer, API) {
  * @returns {PromiseLike<void>}
  */
 export async function refreshData(query = {}) {
-    let API = await login();
+    let API = await ApiWrapper.getInstance();
```
Comment (jakewilson801) [inline lib/etl/scrape.js:111]: `const`

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "outDir": "./dist",
+    "allowJs": true,
+    "target": "es6"
```
Comment (jakewilson801) [inline tsconfig.json:6]: under this I'd highly recommend noImplicitAny true and strictNullChecks true. It's worth it to do it I promise

**Review by jakewilson801 — COMMENTED** [review]
Good luck dude!

File: app.ts [hunk 5]
```diff
@@ -31,6 +35,4 @@ app.use('/api', api_router)
 app.get('/', (req, res)=>{
```
Comment (jakewilson801) [inline app.ts:35]: Two things here Server side rendering and graphQL are big in JS right now. I'd recommend next.js and apollo graphql

Comment (fjaxyu) [reply app.ts:35]: I really like nextjs, it's so good!

**Comment by fjaxyu** [comment]
What's the benefit of installing typescript and node-ts globally instead of just doing it for the project?

**Comment by EcZachly** [comment]
Do you get the ts-node and tsc
command line utilities without installing globally?

**Comment by EcZachly** [comment]
npm start is now
ts-node app.ts

I’m not sure if that runs without a global install.

**Comment by fjaxyu** [comment]
That's true, I don't cli works without it, however the command should still work when using npm start.

**Pull request merged**